bokehjs/src/coffee/models/renderers/glyph_renderer.coffee

```diff
@@ -137,8 +137,11 @@ export class GlyphRendererView extends RendererView
     dtmap = Date.now() - t0
 
     tmask = Date.now()
-    # all_indices and indices are in cdsview subset space
+    # all_indices is in full data space, indices is converted to subset space
+    # either by mask_data (that uses the spatial index) or manually
     indices = @glyph.mask_data(@all_indices)
+    if '' + indices == '' + @all_indices
+      indices = [0...@all_indices.length]
```

Comment:
Is this testing equality by converting to strings? Wouldn't s length comparison suffice? They can't be equal unless they are the same length and (assuming duplicate entries are not possible) they will only be the same length when they are equal.

Reply:
Yes, it is meant to see if mask_data was a no-op. So, I think the length comparison would result in the correct behavior, but the logic would be a little bit wrong. Right now, all of the glyphs that implement _mask_data use the spatial index, which will return the correctly masked data in subset space. So [1, 20, 45] -> [0, 1, 2]. Here the length is the same, but we don't need to convert the indices to subset space. That said, the conversion would simply result in the same [0, 1, 2]. If mask_data does return a shorter array, then it will correctly not run the additional conversion ([1, 20, 45] -> [0, 2] -/> [0, 1], that last arrow would be wrong to do). This code is not ideal because it's not clear that mask_data should convert from full data space to subset space, it just happens to do so.

Reply:
I guess I am worried about performance implications. For scenarios that are currently just on the edge of possible (let's say 50k pts for a concrete plausible number), this means serializing two 50k element lists to strings and comparing on every render? I think that may push "currently just possible" cases into "no longer reasonable" territory. 

When you say "So, I think the length comparison would result in the correct behavior, but the logic would be a little bit wrong" do you mean that the logic is simply indirect and possibly confusing to anyone looking at it? Or that it only happens to be correct in this case, but will be wrong in other cases? If the former, I think that is solved with a comment explaining why the code is doing the comparison

Reply:
Yes, it would be the former, so I agree the length comparison would be the better way to go given the performance cost.